api/deploy.go

```diff
@@ -400,3 +399,77 @@ func deployRebuild(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	}
 	return nil
 }
+
+// title: rollback update
+// path: /apps/{appname}/deploy/rollback/update
+// method: PUT
+// consume: application/x-www-form-urlencoded
+// responses:
+//   200: Rollback updated
+//   400: Invalid data
+//   403: Forbidden
+//   404: Not found
+func deployRollbackUpdate(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	appName := r.URL.Query().Get(":appname")
+	instance, err := app.GetByName(appName)
+	if err != nil {
+		return &tsuruErrors.HTTP{Code: http.StatusNotFound, Message: fmt.Sprintf("App %s not found.", appName)}
+	}
+	canUpdateRollback := permission.Check(t, permission.PermAppUpdateDeployRollback, contextsForApp(instance)...)
+	if !canUpdateRollback {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusForbidden,
+			Message: "User does not have permission to do this action in this app",
+		}
+	}
+	opts := app.DeployOptions{
+		App:  instance,
+		User: t.GetUserName(),
+		Kind: app.DeployRollback,
+	}
+	var imageID string
+	evt, err := event.New(&event.Opts{
+		Target:        appTarget(appName),
+		Kind:          permission.PermAppUpdate,
+		Owner:         t,
+		CustomData:    opts,
```

Comment:
I think this `CustomData` should not be an `app.DeployOptions`, It should probably be created from the request form values with `event.FormToCustomData`